strategy-cache-update-and-refresh/controlled.js

```diff
@@ -0,0 +1,46 @@
+var CACHE = 'cache-update-and-refresh';
+
+if ('serviceWorker' in navigator) {
+  navigator.serviceWorker.onmessage = function (evt) {
+    var message = JSON.parse(evt.data);
+
+    var isRefresh = message.type === 'refresh';
+    var isAsset = message.url.includes('asset');
+    var lastETag = localStorage.currentETag;
+    var isNew =  lastETag !== message.eTag;
+
+    if (isRefresh && isAsset && isNew) {
+      // Escape the first time (when there is no ETag yet)
+      if (lastETag) {
+        // Inform the user about the update
+        notice.hidden = false;
+      }
+      localStorage.currentETag = message.eTag;
```

Comment:
it might be good to explain why are you using `localStorage` instead of `cache` here